db/python/layers/sample.py

```diff
@@ -206,3 +261,17 @@ async def get_history_of_sample(
             )
 
         return rows
+
+    async def batch_upsert_samples(
+        self, samples: SampleBatchUpsert, seqt: SampleSequenceLayer
```

Comment:
This function can instantiate the `SampleSequenceLayer(self.connection)`, I think that would be cleaner than passing it in as a variable.